vllm/sequence.py

```diff
@@ -762,18 +763,23 @@ def init_multi_step_from_lookahead_slots(self, num_lookahead_slots: int,
             assert num_lookahead_slots + 1 == num_scheduler_steps or is_prefill
             self.init_multi_step(num_steps=num_lookahead_slots + 1)
 
-    def get_last_latency(self, now: float) -> float:
+    def set_last_token_time(self, now: float) -> None:
         """Sets the last token time for Request level timings."""
         # If still in prefill phase, raise Error.
         if self.is_prefill():
             raise ValueError(
                 "seq_group.get_last_latency() should not be called "
                 "if the seq_group is in prefill phase.")
-
-        # Otherwise return token latency.
-        latency = now - self.metrics.last_token_time
+        self.last_token_latency = now - self.metrics.last_token_time
         self.metrics.last_token_time = now
-        return latency
+
+    def get_last_token_latency(self) -> float:
+        """Returns the latency of the last token."""
+        if self.is_prefill():
+            raise ValueError(
+                "seq_group.get_last_token_latency() should not be called "
+                "if the seq_group is in prefill phase.")
```

Comment:
Nit: now I think about it, we should make this check (as well as in `set_last_token_time`) `assert ...` instead of `if ... raise`  WDYT?

Reply:
["Assertions can be disabled if you use the -O flag or run from .pyo files instead of .pyc files, so they should not be part of regular error detection."](https://stackoverflow.com/a/40183165)

Due to above reason, my suggestion is to use `if ... raise` (current behavior) let me know if I am missing your intent.

Reply:
The way I think about this particular check here is that this is internal to vLLM versus something that could be possibily caused by a user input.

Quoting from the same SO answer: `Assertions are generally reserved for "I swear this cannot happen" issues that seem to happen anyway.` Here `set_last_token_time` and `get_last_token_latency` are not supposed to be called during prefill phase, so we're not trying to catch the case `if self.is_prefill()`, but rather doing a sanity check, and raising a `ValueError` doesn't really make sense IMO.

Of course this is more of a code style nit - I'm okay with leaving it as is too!

Reply:
Sounds good, switched it to assert.